surfactant/infoextractors/file_decompression.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import tarfile
+import tempfile
+import zipfile
+from queue import Queue
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+import surfactant.plugin
+from surfactant import ContextEntry
+from surfactant.sbomtypes import SBOM, Software
+
+
+def is_compressed(filename):
+    # Map file extensions to their compression formats
+    compression_formats = {
+        ".zip": "zip",
+        ".tar": "tar",
+        ".tar.gz": "tar.gz",
+        ".tar.bz2": "tar.bz2",
+        ".tar.xz": "tar.xz",
+    }
+    for ext, fmt in compression_formats.items():
+        if filename.endswith(ext):
+            return fmt
+    return None
+
+
+@surfactant.plugin.hookimpl
+def extract_file_info(
+    sbom: SBOM, software: Software, filename: str, filetype: str, context: Queue[ContextEntry]
```

Comment:
```suggestion
    sbom: SBOM, software: Software, filename: str, filetype: str, context: "Queue[ContextEntry]"
```

I think quoting the type will fix the Python 3.8 TypeError in the CI tests.

Reply:
Yaaassss, that did it!